Source/MoreDotNet.Test/Extensions/Collections/EnumerableExtensions/ForEachTests.cs

```diff
@@ -0,0 +1,36 @@
+﻿namespace MoreDotNet.Tests.Extensions.Collections.EnumerableExtensions
+{
+    using System;
+    using System.Collections.Generic;
+    using MoreDotNet.Extensions.Collections;
+    using Xunit;
+
+    public class ForEachTests
+    {
+        [Fact]
+        public void ForEach_ShouldAddToResultSet()
+        {
+            List<int> actual = new List<int>();
+            int[] input = new[] { 1 };
+            EnumerableExtensions.ForEach(input, (_) => actual.Add(_));
+            Assert.True(actual.Contains(input[0]));
+        }
+
+        [Fact]
+        public void ForEach_ShouldThrowOnNullItems()
+        {
+            Assert.Throws(
+                typeof(ArgumentNullException),
+                () => EnumerableExtensions.ForEach<int>(null, null));
```

Comment:
When asserting for exceptions, please use the generic overload:
Assert.Throws<T>
Example:
Assert.Throws(typeof(ArgumentNullException), () => EnumerableExtensions.ForEach<int>(null, null));
can be written as:
Assert.Throws<ArgumentNullException>(() => EnumerableExtensions.ForEach<int>(null, null));